test/run_pass/test_spi_buffer_async.js

```diff
@@ -15,25 +15,22 @@
 
 var assert = require('assert');
 var spi = require('spi');
+var pin = require('tools/systemio_common').pin;
+var checkError = require('tools/systemio_common').checkError;
 
-var configuration = {};
-
-if (process.platform === 'linux') {
-  configuration.device = '/dev/spidev0.0';
-} else if (process.platform === 'nuttx' || process.platform === 'tizenrt') {
-  configuration.bus = 1;
-} else {
-  assert.fail();
-}
+var configuration = {
+    device: pin.spi1, // for Linux
+    bus: pin.spi1, // for Tizen, TizenRT and Nuttx
+};
 
 // Buffer test
 var spi1 = spi.open(configuration, function(err) {
-  assert.equal(err, null);
+  checkError(err);
   var data = 'Hello IoTjs';
   var tx = new Buffer(data);
 
   spi1.transfer(tx, function(err, rx) {
-    assert.equal(err, null);
+    checkError(err);
     assert.equal(rx.length, 11);
```

Comment:
`var len = 'Hello IoTjs'.length;`. Using a cache for the length seems better instead of `11`.

a) assert.equal(rx.length, 11);
b) for (var i = 0; i < 11; i++) {